include/windows/osd.h

```diff
@@ -0,0 +1,298 @@
+	return 0;
+}
+
+static inline char* strsep(char** stringp, const char* delim)
+{
+	char* ptr = *stringp;
+	char* p;
+
+	p = ptr ? strpbrk(ptr, delim) : NULL;
+
+	if(!p)
+		*stringp = NULL;
+	else
+	{
+		*p = 0;
+		*stringp = p + 1;
+	}
+
+	return ptr;
+}
+
+static inline int pthread_join(pthread_t thread, void** exit_code)
+{
+	if (WaitForSingleObject(thread, INFINITE) == WAIT_OBJECT_0)
+	{
+		if (exit_code)
+		{
+			DWORD ex = 0;
+			GetExitCodeThread(thread, &ex);
+			*exit_code = (void*)(uint64_t)ex;
+		}
+		CloseHandle(thread);
+		return 0;
+	}
+
+	return -1;
+}
+
+typedef struct fi_thread_arg
+{
+	void* (*routine)(void*);
+	void* arg;
+} fi_thread_arg;
+
+static DWORD WINAPI fi_thread_starter(void* arg)
+{
+	fi_thread_arg data = *(fi_thread_arg*)arg;
+	free(arg);
+	return (DWORD)(uint64_t)data.routine(data.arg);
+}
+
+static inline int pthread_create(pthread_t* thread, void* attr, void *(*routine)(void*), void* arg)
+{
+	attr; /* suppress warning */
+	fi_thread_arg* data = (fi_thread_arg*)malloc(sizeof(*data));
+	data->routine = routine;
+	data->arg = arg;
+	DWORD threadid;
+	*thread = CreateThread(0, 0, fi_thread_starter, data, 0, &threadid);
+	return *thread == 0;
+}
+
+struct iovec
+{
+	void *iov_base; /* Pointer to data.  */
+	size_t iov_len; /* Length of data.  */
+};
+
+#define __attribute__(x)
+
+static inline int poll(struct pollfd *fds, int nfds, int timeout)
+{
+	return WSAPoll(fds, nfds, timeout);
+}
+
+static inline int ofi_memalign(void **memptr, size_t alignment, size_t size)
+{
+	*memptr = _aligned_malloc(alignment, size);
+	return *memptr != 0;
+}
+
+static inline void ofi_freealign(void *memptr)
+{
+	_aligned_free(memptr);
+}
+
+static inline ssize_t ofi_read_socket(int fd, void *buf, size_t count)
+{
+	return recv(fd, (char*)buf, count, 0);
+}
+
+static inline ssize_t ofi_write_socket(int fd, const void *buf, size_t count)
+{
+	return send(fd, (const char*)buf, count, 0);
+}
+
+static inline int ofi_close_socket(int socket)
+{
+	return closesocket(socket);
+}
+
+static inline int fi_fd_nonblock(int fd)
+{
+	u_long argp = 1;
+	return ioctlsocket(fd, FIONBIO, &argp) ? -WSAGetLastError() : 0;
+}
+
+static inline int fi_wait_cond(pthread_cond_t *cond, pthread_mutex_t *mut, int timeout)
+{
+	return !SleepConditionVariableCS(cond, mut, (DWORD)timeout);
+}
+
+/* stubs for Linux only functions */
+static inline void *mremap(void *old_address, size_t old_size, size_t new_size, int flags)
+{
+	return MAP_FAILED;
+}
+
+static inline int shm_open(const char *name, int oflag, mode_t mode)
+{
+	return -1;
+}
+
+static inline int shm_unlink(const char *name)
+{
+	return -1;
+}
+
+static inline void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset)
+{
+	return MAP_FAILED;
+}
+
+static inline int munmap(void *addr, size_t length)
+{
+	return -1;
+}
+
+static inline int ftruncate(int fd, off_t length)
+{
+	return -1;
+}
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* _FI_WIN_OSD_H_ */
+
```

Comment:
Can any of the definitions in this header file move into more appropriate headers?  For example, move the pthread stuff into windows/pthread.h, so the definitions are not split between two files.